src/patternfly/components/AppLauncher/examples/AppLauncher.md

```diff
@@ -197,6 +197,112 @@ import './AppLauncher.css'
 {{/app-launcher}}
 ```
 
+```hbs title=Favorites
+{{#> app-launcher id="app-launcher-sections-dividers-icons-and-external-links" app-launcher--IsExpanded="true" app-launcher--IsGrouped="true"}}
+  {{#> app-launcher-menu}}
+    {{#> app-launcher-menu-input}}
+      {{#> form-control controlType="input" input="true" form-control--attribute=(concat 'type="search" placeholder="Filter by name..." id="' id '-text-input" name="textInput1"')}}
+        {{/form-control}}
+    {{/app-launcher-menu-input}}
+    {{#> app-launcher-group}}
+      {{#> app-launcher-group-title}}
+        Favorites
+      {{/app-launcher-group-title}}
+      <ul>
+        {{#> app-launcher-menu-item app-launcher-menu-item--type="li" app-launcher-menu-item--modifier="pf-m-wrapper pf-m-external pf-m-favorite" app-launcher-menu-item--attribute='href="#"'}}
+          {{#> app-launcher-menu-item app-launcher-menu-item--modifier="pf-m-link" app-launcher-menu-item-favorite-link--attribute='href="#" target="_blank"'}}
```

Comment:
Should both the `li` and the child `a` have the same class with just a different modifier? Shouldn't they just have different classes via BEM?

Reply:
`app-launcher-menu-item-favorite-link--attribute` wasn't added so `app-launcher-menu-item-favorite-link--attribute='href="#"` isn't adding the href attribute.

Reply:
> Should both the li and the child a have the same class with just a different modifier? Shouldn't they just have different classes via BEM?

I wondered the same thing. I started out building them as their own BEM elements but realized they're basically just items and have the same basic style as the original item, with a few adjustments in padding and background color mostly. They all share some styles of the original item CSS, with some changes. I ended up with this, and it felt better than creating unique elements.

That said I pushed a unique commit in https://github.com/patternfly/patternfly-next/pull/2428/commits/c3a0f0a3f09c9369fd861645bd89d2415e09ca1d where I did the converstion of changing the outer `.pf-c-app-launcher__menu-item.pf-m-wrapper` to `.pf-c-app-launcher__menu-item-wrapper` and updated the rest of the styles to work. I think it makes things a bit more complicated but have a look and let me know what you think.

Reply:
ya it's definitely not as straight forward. My main concern is how this could cause confusion because they are not siblings. I suppose it's just down to how strict are we when following BEM and how concerned are we that this could be implemented incorrectly? Will this have any impact on the React implementation?

Reply:
Are you referring to how the structure is `__menu-item-wrapper` -> `__menu-item`, when we normally name classes so that `__menu-item[-whatever]` would be a child of `__menu-item`?

Reply:
ya exactly.

Reply:
Ah gotcha, I renamed it to `__menu-wrapper`. Do you think this approach of it being it's own BEM element is preferred to it being `.pf-c-app-launcher__menu-item.pf-m-wrapper` (or use some other word rather than "wrapper" in the modifier or BEM name)? I think the modifier works as well. I'm curious what @mattnolting and @christiemolloy think about that, too.